File: Module 2/Week 1/6. Leap - Year/Leap-Year/src/LeapYear.java
import java.util.Scanner;

public class LeapYear {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Input year: ");
        int year = sc.nextInt();
        if (year % 4 == 0 && year % 100 != 0) {
            System.out.println(year + " Is a leap year");
        } else if (year % 100 == 0 && year % 400 == 0) {
            System.out.println(year + " Is a leap year");
        } else {
            System.out.println(year + " Not a leap year");
        }
    }
}

File: Module 1/Week2/8. Leap-year/leap-year.js
function leap_year() {
    let year = document.getElementById("year").value;
    let a = true;

    if (year % 4 === 0) {
        if (year % 100 !== 0) {
            a = true;
        }
        else {
            a = false;
        }
    }
    else if (year % 100 === 0) {
        if (year % 400 !== 0) {
            a = false;
        }
        else {
            a = true;
        }
    }
    else if (year % 100 === 0) {
        if (year % 400 === 0) {
            a = true;
        }
        else {
            a = false;
        }
    }
    else {
        a = false;
    }

    if (a===true){
        document.getElementById("result").innerHTML = "Leap Year";
    }
    else{
        document.getElementById("result").innerHTML = "Not Leap Year";
    }







}